buildSrc/src/main/groovy/org/elasticsearch/gradle/precommit/LoggerUsageTask.groovy

```diff
@@ -38,23 +38,32 @@ public class LoggerUsageTask extends LoggedExec {
 
     private FileCollection classpath;
 
-    private List<File> classDirectories;
+    private FileCollection classDirectories;
 
     public LoggerUsageTask() {
         project.afterEvaluate {
             dependsOn(classpath)
             description = "Runs LoggerUsageCheck on ${classDirectories}"
             executable = new File(project.javaHome, 'bin/java')
             if (classDirectories == null) {
-                classDirectories = []
-                if (project.sourceSets.findByName("main") && project.sourceSets.main.output.classesDir.exists()) {
-                    classDirectories += [project.sourceSets.main.output.classesDir]
+                // Default to main and test class files
+                List files = []
+                // But only if the source sets that will make them exist
+                if (project.sourceSets.findByName("main")) {
+                    files.add(project.sourceSets.main.output.classesDir)
                     dependsOn project.tasks.classes
                 }
-                if (project.sourceSets.findByName("test") && project.sourceSets.test.output.classesDir.exists()) {
-                    classDirectories += [project.sourceSets.test.output.classesDir]
+                if (project.sourceSets.findByName("test")) {
+                    files.add(project.sourceSets.test.output.classesDir)
                     dependsOn project.tasks.testClasses
                 }
+                /* In an extra twist, it isn't good enough that the source set
```

Comment:
The language here is confusing: "...that the source set empty source sets won't..."